src/session-pool.ts

```diff
@@ -485,6 +491,9 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
     });
 
     this._traces = new Map();
+    if (!this._observabilityOptions) {
```

Comment:
Why do we need this if check. We should simply override the observability option ?

Reply:
Not really, I was being cautious in case someone had passed in multiple observability configurations one for the explicitly created SessionPool and then one for the Database separately. It doesn't hurt to be conservative at the beginning before these libraries have been put into heavy usage.

Reply:
But we don't allow customers to pass multiple observability configurations right ? They should be able to set it only once via spannerOptions. 

We also modified the _observabilityOptions object to add _ to represent private variable.